integration/src/test/java/org/apache/iotdb/db/integration/IoTDBSelectCompareExpressionIT.java

```diff
@@ -0,0 +1,452 @@
+        assertEquals(_double.get(i) == 5, bool);
+
+        bool = Boolean.parseBoolean(resultSet.getString(7));
+        assertEquals(_double.get(i) != 5, bool);
+      }
+    } catch (SQLException throwable) {
+      fail(throwable.getMessage());
+    }
+
+    try (Connection connection = EnvFactory.getEnv().getConnection();
+        Statement statement = connection.createStatement()) {
+      ResultSet resultSet =
+          statement.executeQuery("select s5==true, s5!=true, s5==false, s5!=false from root.sg.d1");
+      int columnCount = resultSet.getMetaData().getColumnCount();
+      assertEquals(1 + 4, columnCount);
+
+      boolean bool;
+      for (int i = 0; i < time.size(); ++i) {
+        resultSet.next();
+
+        bool = Boolean.parseBoolean(resultSet.getString(2));
+        assertEquals(_bool.get(i) == true, bool);
+
+        bool = Boolean.parseBoolean(resultSet.getString(3));
+        assertEquals(_bool.get(i) != true, bool);
+
+        bool = Boolean.parseBoolean(resultSet.getString(4));
+        assertEquals(_bool.get(i) == false, bool);
+
+        bool = Boolean.parseBoolean(resultSet.getString(5));
+        assertEquals(_bool.get(i) != false, bool);
+      }
+    } catch (SQLException throwable) {
+      fail(throwable.getMessage());
+    }
+  }
+
+  @Test
+  public void testCompareDifferentType() {
+    try (Connection connection = EnvFactory.getEnv().getConnection();
+        Statement statement = connection.createStatement()) {
+      ResultSet resultSet =
+          statement.executeQuery(
+              "select s1>=s2, s1<=s2, s1>s3, s1<s3, s1==s4, s1!=s4 from root.sg.d1");
+      int columnCount = resultSet.getMetaData().getColumnCount();
+      assertEquals(1 + 6, columnCount);
+
+      boolean bool;
+      for (int i = 0; i < time.size(); ++i) {
+        resultSet.next();
+
+        bool = Boolean.parseBoolean(resultSet.getString(2));
+        assertEquals(_int.get(i) >= _long.get(i), bool);
+
+        bool = Boolean.parseBoolean(resultSet.getString(3));
+        assertEquals(_int.get(i) <= _long.get(i), bool);
+
+        bool = Boolean.parseBoolean(resultSet.getString(4));
+        assertEquals(_int.get(i) > _float.get(i), bool);
+
+        bool = Boolean.parseBoolean(resultSet.getString(5));
+        assertEquals(_int.get(i) < _float.get(i), bool);
+
+        bool = Boolean.parseBoolean(resultSet.getString(6));
+        assertEquals((double) _int.get(i) == _double.get(i), bool);
+
+        bool = Boolean.parseBoolean(resultSet.getString(7));
+        assertEquals((double) _int.get(i) != _double.get(i), bool);
+      }
+    } catch (SQLException throwable) {
+      fail(throwable.getMessage());
+    }
+
+    try (Connection connection = EnvFactory.getEnv().getConnection();
+        Statement statement = connection.createStatement()) {
+      ResultSet resultSet =
+          statement.executeQuery(
+              "select s2>=s3, s2<=s3, s2>s4, s2<s4, s2==s1, s2!=s1 from root.sg.d1");
```

Comment:
Could you add a more complex test case like 'select s1 * 2 <= s2 + 1 * 10' and so on?